build-system/pr-check/remote-tests.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script kicks off the unit and integration tests on Sauce Labs.
+ * This is run during the CI stage = test; job = remote tests.
+ */
+
+const {
+  downloadBuildOutput,
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  startSauceConnect,
+  stopSauceConnect,
+  timedExecOrDie: timedExecOrDieBase} = require('./utils');
+const {determineBuildTargets} = require('./build-target');
+const {isTravisPullRequestBuild} = require('../travis');
+
+const FILENAME = 'remote-tests.js';
+const timedExecOrDie =
+  (cmd, unusedFileName) => timedExecOrDieBase(cmd, FILENAME);
+
+function main() {
+  const startTime = startTimer(FILENAME, FILENAME);
+  const buildTargets = determineBuildTargets();
+  printChangeSummary(FILENAME);
+  downloadBuildOutput(FILENAME);
+  startSauceConnect(FILENAME);
+
+  if (!isTravisPullRequestBuild()) {
+    timedExecOrDie('gulp test --unit --nobuild --saucelabs_lite');
+    timedExecOrDie('gulp test --integration --nobuild --compiled --saucelabs');
```

Comment:
This is running integration tests in `--compiled` mode (which means it expects `gulp dist --fortesting` to have been run at some point prior). The implementation of `build.js` seems to be breaking that assumption.

One thing you could do to fix this is to run `gulp dist --fortesting` after `gulp build` in `build.js` in the push build case. It will increase the running time of the `Build` job alone to ~30 minutes. We can revisit this in a follow up PR to optimize the timing of push builds.

Reply:
What if we ran `gulp test --fortesting` in the remote tests job? 

We're running `gulp dist --fortesting --single_pass --psuedonames` and `gulp dist --fortesting --noextensions` in the dist jobs for push builds, so it wouldn't be breaking a pattern. It would also allow for the test stage to start sooner.

Reply:
I assume you meant...

> What if we ran `gulp dist --fortesting` in the remote tests job?

Yes, that will certainly solve the problem, since it's just the remote tests that run in compiled mode today. This SGTM.

Reply:
Yes gulp dist! Okay great.